include/moqt/core/transport.h

```diff
@@ -0,0 +1,226 @@
+        Status GetStatus();
+
+        // --------------------------------------------------------------------------
+        // Metrics
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Connection metrics for server accepted connections
+         *
+         * @details Connection metrics are updated real-time and transport quic metrics on
+         *      Config::metrics_sample_ms period
+         */
+        std::map<TransportConnId, ConnectionMetrics> connection_metrics_;
+
+      private:
+        // -------------------------------------------------------------------------------------------------
+        // Transport Delegate/callback functions
+        // -------------------------------------------------------------------------------------------------
+
+        void on_new_data_context([[maybe_unused]] const TransportConnId& conn_id,
+                                 [[maybe_unused]] const DataContextId& data_ctx_id) override
+        {
+        }
+
+        void on_connection_status(const TransportConnId& conn_id, const TransportStatus status) override;
+        void on_new_connection(const TransportConnId& conn_id, const TransportRemote& remote) override;
+        void on_recv_stream(const TransportConnId& conn_id,
+                            uint64_t stream_id,
+                            std::optional<DataContextId> data_ctx_id,
+                            const bool is_bidir = false) override;
+        void on_recv_dgram(const TransportConnId& conn_id, std::optional<DataContextId> data_ctx_id) override;
+
+        // -------------------------------------------------------------------------------------------------
+        // End of transport delegate/callback functions
+        // -------------------------------------------------------------------------------------------------
+
+        struct ConnectionContext
+        {
+            TransportConnId conn_id;
+            std::optional<uint64_t> ctrl_data_ctx_id;
+            bool setup_complete{ false }; ///< True if both client and server setup messages have completed
+            uint64_t client_version{ 0 };
+            std::optional<messages::MoqtMessageType>
+              ctrl_msg_type_received; ///< Indicates the current message type being read
+
+            uint64_t current_subscribe_id{ 0 }; ///< Connection specific ID for subscribe messages
+
+            /// Track namespace/name by received subscribe IDs
+            /// Used to map published tracks to subscribes in client mode
+            std::map<uint64_t, std::pair<uint64_t, uint64_t>> recv_sub_id;
+
+            /// Tracks by subscribe ID
+            std::map<uint64_t, std::shared_ptr<BaseTrackHandler>> tracks_by_sub_id;
+
+            /// Publish tracks by namespace and name. map[track namespace][track name] = track delegate
+            std::map<uint64_t, std::map<uint64_t, std::shared_ptr<BaseTrackHandler>>> pub_tracks_by_name;
```

Comment:
should we use FullTrackName instead ?

Reply:
Again, this is not part of the public API as these are private members used by the implementation.